packages/nimble-components/src/table-column/enum-base/models/mapping-icon-config.ts

```diff
@@ -1,28 +1,36 @@
-import { html, type ViewTemplate } from '@microsoft/fast-element';
+import { html, when, type ViewTemplate } from '@microsoft/fast-element';
 import type { IconSeverity } from '../../../icon-base/types';
 import { MappingConfig } from './mapping-config';
 
 export interface IconView {
     severity: IconSeverity;
     text?: string;
 }
-const createIconTemplate = (icon: string): ViewTemplate<IconView> => html`
-    <${icon}
-        title="${x => x.text}"
-        role="img"
-        aria-label="${x => x.text}"
-        severity="${x => x.severity}"
-        class="no-shrink"
-    >
-    </${icon}>`;
+const createIconTemplate = (
+    icon: string | undefined
+): ViewTemplate<IconView> => html`
+    ${when(
+        () => icon,
```

Comment:
In this case the evaluation of icon is static at the call of createIconTemplate, not dynamically determined (i.e. we don't check the context of what the template is bound to with `x => x.icon`, iit is always statically determined no matter what we are bound to `() => icon`).

Since it is static, can we instead avoid the extra runtime logic in the template?

```ts
const emptyTemplate = html<IconView>` `; // I think you at least need a space in there or you get a runtime error.

const createIconTemplate = (icon) => icon === undefined ? emptyTemplate : html<IconView>`
        <${icon}
            title="${x => x.text}"
            role="img"
            aria-label="${x => x.text}"
            severity="${x => x.severity}"
            class="no-shrink"
        >
        </${icon}>
```

Has the minor added benefit that all empty icons get to re-use the same template instance.